Added a doc for IDE setup

Samyoul opened the pull request:
Based on https://notes.status.im/9Gr7kqF8SzC_SmYK0eB7uQ?view, hopefully this document can help some poor soul in the future. I have a little more work to do but I'm interested in whether this kind of document would be welcome in the repo.

After `make clean` I found that I had a lot of reset up to do and it took me a long time to find all the information to recreate the setup. This document will be invaluable for at least a future me, so maybe it could be useful to others.

To easily see the document see https://github.com/status-im/status-react/blob/55ccda27340f97f874c623abd3c9072989ccd199/doc/IDE_SETUP.md

---

Todos for this PR:

- [x] Add a table of contents
- [x] Tidy up the Troubleshooting parts
  - [x] Remove references to things I tried and detail from a this could go wrong, this is how to fix it perspective
- [x] Add all images to the repo to prevent future 404s
- [x] Some images are huge others are small, resize big images so the document is easier to read
- [x] Add more detailed description of connecting and using REPL with IntelliJ
- [x] Link to other parts in the docs that are relevant

Samyoul committed:
Added a doc for IDE setup

Based on https://notes.status.im/9Gr7kqF8SzC_SmYK0eB7uQ?view, hopefully this document can help some poor soul in the future. I have a little more work to do but I'm interested in whether this kind of document would be welcome in the repo

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | c1080097 | [#1](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-11888/1/) | 2021-03-17 00:00:29 | ~10 min | `android-e2e` | [:package:`apk`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-210316-235002-c10800-pr-x86.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-210316-235002-c10800-pr-x86.apk)|

flexsurfer commented:
i guess its WIP ?

Samyoul commented:
Yes let me rename it

shivekkhurana commented on doc/IDE_SETUP.md:
````diff
@@ -0,0 +1,170 @@
+=> nil
+```
+
+Or
+
+```shell
+prn 1
+```
+
+Which should output
+
+```shell
+prn 1
+=> 1
+=> #function[clojure.core/prn]
+```
````
This doesn't sound right `prn 1` is not valid without the parenthesis

Samyoul replied:
It works on my machine, you can see in the screenshot below. But it is a bit redundant so I've removed it from the docs

shivekkhurana commented on doc/IDE_SETUP.md:
````diff
@@ -0,0 +1,170 @@
+I can connect to REPL inside the IDE, but the code doesn't seem to be able to send to REPL. But when trying to load any code into REPL I get a `Cannot load Clojure form into ClojureScript REPL`.
+
+My process:
+
+Open IDEA, not yet connected to REPL and not running a terminal with `make run-clojure`.
+
+![](https://notes.status.im/uploads/upload_a2abb27acbee62a0ca9ffdc41d2219dc.png)
+
+Attempting to connect to REPL without `make run-clojure` gives the expected error. After a terminal is running `make run-clojure`, connection works fine.
+
+Entering simple functions gives a response.
+
+![](https://notes.status.im/uploads/upload_9b2848993feb4181ad8fee96fc0dd6e9.png)
+
+However trying to do any thing REPL-ly with the code always gives a "*Cannot load Clojure form into ClojureScript REPL*" error. 
+
+![](https://notes.status.im/uploads/upload_d29b3ded54afdfddf0a2cb09bfdbd121.png)
+
+![](https://notes.status.im/uploads/upload_c07a7805e1e7c954e3d3698063410bb5.png)
+
+---
+
+### Things I've also tried
+
+I've done some Googling to see what could be going wrong.
+
+
+https://github.com/cursive-ide/cursive/issues/1285 This issue suggests switching the file to `cljc` which I did out of curiosity, it connects better but it doesn't really work.
+
+![](https://notes.status.im/uploads/upload_79bca0edc0c1841a7cbdcec50a4f90eb.png)
+
+Obviously this isn't ideal even if it worked perfectly, I presume that the file types are important and converting them back and forth is a terrible work flow.
+
+I tried converting the files to `clj`, just to see if that helped, which gave an interesting error:
+
+```bash=
+Loading src/status_im/reloader.clj... 
+Syntax error (FileNotFoundException) compiling at (src/status_im/reloader.clj:1:1).
+Could not locate status_im/ui/components/react__init.class, status_im/ui/components/react.clj or status_im/ui/components/react.cljc on classpath. Please check that namespaces with dashes use underscores in the Clojure file name.
+```
````
Are these two sections needed since the sections above show how to connect to the repl?

Samyoul replied:
Yeah, I've removed this section

shivekkhurana reviewed (changes requested):
Some sections can be removed.

Samyoul committed:
Added ToC, Tidied Troubleshooting, Added images to avoid future 404s

Removed superfluous check

Adjusted images to be 50% the page width

Image width 75%

Finished off REPL section

Signed-off-by: Samuel Hawksby-Robinson <samuel@samyoul.com>

shivekkhurana commented:
@Samyoul Please feel free to merge this (we don't use github's merge functionality, but the [merge script instead](https://github.com/status-im/status-react/tree/develop/doc#merging-approved-prs)).

Samyoul commented:
That's an interesting merge process.
